### PR TITLE
BUG Fix error when running on ss 3.7

### PR DESCRIPTION
**Pull request by tractorcow**

### CODE REVIEW

File: src/SilverStripe/BehatExtension/Context/BasicContext.php [hunk 1]
```diff
@@ -843,7 +843,7 @@ public function theTextBeforeAfter($textBefore, $order, $textAfter, $element) {
 	public function iWaitXUntilISee($wait, $selector) {
 		$page = $this->getSession()->getPage();
 
-		$this->spin(function($page) use ($page, $selector){
```
Comment (tractorcow) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: Error was ambiguous value of $page due to it both inheriting from parent context as well as being an arg.

File: src/SilverStripe/BehatExtension/Context/FixtureContext.php [hunk 2]
```diff
@@ -342,7 +342,7 @@ public function stepUpdateRecordState($type, $id, $state) {
 				\Versioned::reading_stage('Live');
 				$clone = clone $obj;
 				$clone->delete();
-				\Versioned::reading_stage($oldMode);
+				\Versioned::set_reading_mode($oldMode);
```
Comment (tractorcow) [inline src/SilverStripe/BehatExtension/Context/FixtureContext.php]: $oldMode was a mode not a stage, so reading_stage() threw errors.

File: src/SilverStripe/BehatExtension/Context/BasicContext.php (hunk 1 above)
Comment (robbieaverill) [reply src/SilverStripe/BehatExtension/Context/BasicContext.php]: Yep, that looks dubious and removing it makes sense in the context of these step definitions

**Comment by robbieaverill** [comment]
Build failure is PHP 5.3 related. We should remove it from Travis.

**Pull request merged**